zkevm/src/circuit.rs

```diff
@@ -242,3 +249,74 @@ impl TargetCircuit for PoseidonCircuit {
         hash_rows
     }
 }
+
+pub struct ByteCodeCircuit {}
+impl TargetCircuit for ByteCodeCircuit {
+    type Inner = ByteCodeCircuitImpl<Fr>;
+
+    fn name() -> String {
+        "ByteCode".to_string()
+    }
+
+    fn empty() -> Self::Inner {
+        let r = tests::get_randomness();
+
+        let bytecodes: Vec<UnrolledBytecode<Fr>> = vec![bytecode_unroller::unroll(vec![], r)];
+        let circuit = ByteCodeCircuitImpl::<Fr> {
+            bytecodes: bytecodes,
+            size: 2usize.pow(*DEGREE as u32),
+            randomness: r,
+        };
+
+        circuit
+    }
+
+    fn from_block_result(block_result: &BlockResult) -> anyhow::Result<(Self::Inner, Vec<Vec<Fr>>)>
+    where
+        Self: Sized,
+    {
+        let witness_block = block_result_to_witness_block::<Fr>(block_result)?;
+        let r = witness_block.randomness;
+        let byte_codes = witness_block
+            .bytecodes
+            .values()
+            .into_iter()
+            .map(|bytecode| bytecode_unroller::unroll(bytecode.clone().bytes, r))
+            .collect();
+
+        let inner = ByteCodeCircuitImpl::<Fr> {
+            bytecodes: byte_codes,
+            size: 2usize.pow(*DEGREE as u32),
+            randomness: r,
+        };
+
+        // skip setting instance info as current aggr circuit don't support instance column
+        // let num_rows = 1 << *DEGREE;
+        // const NUM_BLINDING_ROWS: usize = 7 - 1;
+        // let instance = vec![vec![r; num_rows - NUM_BLINDING_ROWS]];
+        Ok((inner, vec![]))
+    }
+
+    fn estimate_rows(block_result: &BlockResult) -> usize {
+        if let Ok(witness_block) = block_result_to_witness_block::<Fr>(block_result) {
```

Comment:
i suggest `block_result_to_witness_block::<Fr>(block_result).unwrap()`.  block_result_to_witness_block should always success or else it is a bug we should fix

Reply:
not see the error case, changed to unwrap [c93e757](https://github.com/scroll-tech/common-rs/pull/61/commits/c93e757d6c57a85dac5a45a693651a5f4d11809c)